frontend/test/__support__/server-mocks/search.ts

```diff
@@ -1,8 +1,11 @@
 import fetchMock from "fetch-mock";
 
-import type { CollectionItem, SearchResult } from "metabase-types/api";
+import type { CollectionItem } from "metabase-types/api";
+import type { WeakSearchResult } from "metabase-types/entities/search";
 
-export function setupSearchEndpoints(items: (CollectionItem | SearchResult)[]) {
+export function setupSearchEndpoints(
+  items: (CollectionItem | WeakSearchResult)[],
```

Comment:
I don't think we should be changing the type of the server mocks here. the server returns what it returns, the type should reflect that.